thrift/tchannel.thrift

```diff
@@ -0,0 +1,8 @@
+struct ClientLibrary {
```

Comment:
Here's an idea that might help with Meta vs TChannel

Meta is something that either:

 - The application developer can customize
 - Contains optional things.

The TChannel api is something that will be:

 - Implemented by tchannel without user customization
 - Does not contain optional features at all.

For example:

 - Meta health is customized by users so belongs in Meta
 - Meta thriftIdl only applies to thrift encoding so belongs in Meta
 - TChannel library version is not customized and is mandatory so belongs in tchannel introspection.